components/org.wso2.carbon.identity.conditional.auth.functions.elk/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/elk/HttpClientManager.java

```diff
@@ -0,0 +1,227 @@
+                .CONNECTION_POOL_MAX_CONNECTIONS_PER_ROUTE);
+        int defaultMaxConnections = 20;
+        int maxConnections = defaultMaxConnections;
+        int maxConnectionsPerRoute = defaultMaxConnections;
+        try {
+            maxConnections = Integer.parseInt(maxConnectionsString);
+        } catch (NumberFormatException e) {
+            // Ignore. Default value is used.
+        }
+        try {
+            maxConnectionsPerRoute = Integer.parseInt(maxConnectionsPerRouteString);
+        } catch (NumberFormatException e) {
+            // Ignore. Default value is used.
+        }
+
+        ConnectingIOReactor ioReactor;
+        try {
+            ioReactor = new DefaultConnectingIOReactor();
+        } catch (IOReactorException e) {
+            throw new FrameworkException("Error while creating ConnectingIOReactor", e);
+        }
+        PoolingNHttpClientConnectionManager poolingHttpClientConnectionManager = new
+                PoolingNHttpClientConnectionManager(ioReactor);
+        // Increase max total connection to 50
+        poolingHttpClientConnectionManager.setMaxTotal(maxConnections);
+        // Increase default max connection per route to 50
+        poolingHttpClientConnectionManager.setDefaultMaxPerRoute(maxConnectionsPerRoute);
+        return poolingHttpClientConnectionManager;
+    }
+
+    public void closeClient(int tenantId) throws IOException {
+
+        CloseableHttpAsyncClient client = clientMap.get(tenantId);
+
+        if (client != null) {
+            clientMap.remove(tenantId);
+            client.close();
+        }
+    }
+
+    private void addSslContext(HttpAsyncClientBuilder builder, String tenantDomain) {
+
+        try {
+            SSLContext sslContext = SSLContexts.custom()
+                    .loadTrustMaterial(ElasticFunctionsServiceHolder.getInstance().getTrustStore())
+                    .build();
+
+            String hostnameVerifierConfig = CommonUtils.getConnectorConfig(ElasticAnalyticsEngineConfigImpl
+                    .HOSTNAME_VERIFIER, tenantDomain);
+            X509HostnameVerifier hostnameVerifier;
+            if (ElasticAnalyticsEngineConfigImpl.HOSTNAME_VERIFIER_STRICT.equalsIgnoreCase(hostnameVerifierConfig)) {
+                hostnameVerifier = SSLConnectionSocketFactory.STRICT_HOSTNAME_VERIFIER;
+            } else if (ElasticAnalyticsEngineConfigImpl.HOSTNAME_VERIFIER_ALLOW_ALL.equalsIgnoreCase(hostnameVerifierConfig)) {
+                hostnameVerifier = SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER;
+            } else {
+                hostnameVerifier = SSLConnectionSocketFactory.STRICT_HOSTNAME_VERIFIER;
+            }
+
+            builder.setSSLContext(sslContext);
+            builder.setHostnameVerifier(hostnameVerifier);
+        } catch (Exception e) {
+            LOG.error("Error while creating ssl context for analytics endpoint invocation in tenant domain: " +
+                    tenantDomain, e);
+        }
+    }
+
```

Comment:
Unwanted extra line.

Reply:
Resolved in https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/110